### ISSUE TITLE
git backend

### ISSUE THREAD

**Issue by vjeantet** [spf13/afero#227]
Hello,

   Is a git repo could be used as backend ?

#question

**Comment by jonfriesen** [comment]
@vjeantet I'm going to give answering your question a shot.

Theoretically you could use Afero to hold git projects. Doing so will require a git library that allows you to inject a filesystem interface. [src-d/go-git](https://github.com/src-d/go-git) supports this concept, unfortunately the interfaces don't line up so integrating them together is a painful.

the src-d team has created their [own virtual filesystem](https://godoc.org/github.com/src-d/go-billy) which can check code out into memory/other file systems which might satisfy your need. Small disclaimer, it's not nearly built out to the level of Afero and doesn't include many of the utilities.

**Comment by c4milo** [comment]
@jonfriesen, are you all still willing to support the billy adapter within Afero?

**Comment by jonfriesen** [comment]
@c4milo no, I no longer use this library and do not have the time. Thanks (and sorry!)
